### ISSUE TITLE
Embedded Activities

### ISSUE THREAD

**Issue by seaneagan** [PaulKinlan/WebIntents#80]
When a client integrates directly with a service, it can embed the UI of the service within its own UI, creating an integrated mashup-like feel.  Such services include sharing widgets, maps, and document/code editors.  Currently if a client uses web intents to integrate with services they lose this ability since the UI of the service always loads in a separate popup window.  To solve this, web intents could be extended to allow for the embedding of activities.

First, a new element could be added, similar to an iframe, except with complete isolation between the parent and child frames, i.e. `window.parent` would be unset within the child frame, and the dom API of the new element would have no "contentWindow" property for the embedding page to access.  The inner html of the element could be displayed when no activity is currently loaded within it.  Something like:

```
<activity id="activity" class="whatever">
  This content will be shown when no activity is currently loaded
</activity>
```

The API to target an activity to be loaded within such an element could be either of the following:
- set the intent's "target" attribute to the element, e.g. `intent.target = document.getElementById("activity")`
- call a method on the element's dom api which is equivalent to `window.navigator.startActivity` except that the activity starts within the element instead of in a separate window

It may be useful to add a "load" event to these elements.  Then clients could hide the elements until they have an activity loaded within them, at which point they could fade them in etc.  Or, they could simply draw the user's attention to the element when an activity is loaded within it, and is thus ready for interaction.

Services should be able to determine when they are being embedded, so that they can optimize their style such as by using a transparent background, or a sparser ui.  This boolean value could be stored in `window.intent.embedded`.

**Comment by PaulKinlan** [comment]
On the subject of this shim - this is likely to never be implement.  I will raise it through the team for the real implementation.

**Comment by seaneagan** [comment]
Agreed, I don't see any way to implement this within the shim.  If there is a better place to discuss the web intents specification as opposed to the shim, let me know.  In the meantime I added some improvements to the issue text above.

**Comment by PaulKinlan** [comment]
So, we haven't got embedded intents but we have inline.  This means that the picker will display the action that the user wants to run.  This is now in the code.
